lib/angular2/shared/sockets/connections.ts

```diff
@@ -2,85 +2,156 @@
+        this.subjects.onAuthenticated.next();
+        this.heartbeater();
+      })
+      // Listen for authentication
+      this.on('unauthorized', (err: any) => {
+        this.subjects.onUnAuthorized.next(err);
+      })
+      // Listen for disconnections
+      this.on('disconnect', (status: any) => this.subjects.onDisconnect.next(status));
+    } else if (this.socket && !this.socket.connected){
+      this.socket.off();
+      this.socket.destroy();
+      delete this.socket;
+      this.connect(token);
     }
-    return this.connections[url];
   }
-
-  public disconnect() {
-    Object.keys(this.connections).forEach((connKey) => {
-      if (this.connections[connKey] && this.connections[connKey].connected) {
-        this.connections[connKey].disconnect();
-      }
-      if (this.connections[connKey].off) {
-        this.connections[connKey].off();
-      }
-    });
-    this.connections = {};
-    this.configured  = false;
+  /**
+   * @method isConnected
+   * @description
+   * This method will return true or false depending on established connections
+   **/
+  public isConnected(): boolean {
+    return (this.socket && this.socket.connected);
   }
-
-  private setupConnection(url: string, token: AccessToken, config: any): void {
-    this.configured = true;
-    console.log('Connected to %s', url);
-    if (token.id) {
-      console.log('Emitting authentication', token.id);
-      this.connections[url].emit('authentication', token);
-    }
-    this.connections[url].on('unauthorized', (res: any) => console.error('Unauthenticated', res));
+  /**
+   * @method on
+   * @description
+   * This method will wrap the original "on" method and run it within the Angular Zone
+   **/
+  public on(event: string, handler: Function): void {
+    this.socket.on(event, (data: any) => this.zone.run(() => handler(data)));
+  }
+  /**
+   * @method emit
+   * @description
+   * This method will wrap the original "on" method and run it within the Angular Zone
+   **/
+  public emit(event: string, data: any): void {
+    this.socket.emit(event, data);
+  }
+  /**
+   * @method removeListener
+   * @description
+   * This method will wrap the original "on" method and run it within the Angular Zone
+   **/
+  public removeListener(event: string, handler: Function): void {
+    this.socket.removeListener(event, handler);
+  }
+  /**
+   * @method disconnect
+   * @description
+   * This will disconnect the client from the server
+   **/
+  public disconnect(): void {
+    this.socket.disconnect();
+  }
+  /**
+   * @method heartbeater
+   * @description
+   * This will keep the connection as active, even when users are not sending
+   * data, this avoids disconnection because of a connection not being used.
+   **/
+  private heartbeater(): void {
     let heartbeater: any = setInterval(() => {
-      if (this.connections[url]) {
-        this.connections[url].emit('lb-ping');
+      if (this.socket && this.socket.connected) {
```

Comment:
Could be, but would not that add an aditional fn call? making the stack of calls larger. Not big deal just wanted to avoid that

Thoughts?